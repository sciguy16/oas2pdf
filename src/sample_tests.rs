use crate::{args::Args, do_run, test::rand_str};
use std::path::Path;

fn run_sample_test<P: AsRef<Path>>(input: P) -> String {
    let typst_path = format!("/tmp/sample-typst-out-{}", rand_str());
    let mut args = Args {
        out: Some((&typst_path).into()),
        typst: true,
        input: input.as_ref().into(),
        save_template: None,
        template: None,
    };
    do_run(&args).unwrap();
    // Store the typst file for snapshot testing
    let rendered = std::fs::read_to_string(&typst_path).unwrap();

    // try rendering the PDF to check for syntax errors
    args.typst = false;
    do_run(&args).unwrap();

    std::fs::remove_file(typst_path).unwrap();
    rendered
}

#[test]
fn docker_api() {
    insta::assert_snapshot!(run_sample_test("samples/docker.yaml"));
}
