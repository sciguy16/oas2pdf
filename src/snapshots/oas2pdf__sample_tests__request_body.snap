---
source: src/sample_tests.rs
expression: "run_sample_test(\"samples/request-body.yaml\")"
---
#let title = [ Several Methods API ]

#let numbering-limited(max-level, schema) = (..numbers) => {
  if numbers.pos().len() <= max-level {
    numbering(schema, ..numbers)
  }
}
#set heading(numbering: numbering-limited(3, "1."))
#let appendix(body) = {
  set heading(numbering: numbering-limited(1, "A."), supplement: [Appendix])
  counter(heading).update(0)
  body
}

#set page(
  header: align(
    right + horizon,
    title
  ),
  numbering: "1/1"
)

#align(horizon)[
	#align(center, text(27pt)[
	  *#title*
	])
	#align(center, text(24pt)[ A simple API with a request body ])
]

#pagebreak()

#outline(
	depth: 2,
	indent: auto,
)


#pagebreak()
= Container

== /post

=== POST - PostThings
Post things

Post things



- application/json: 

- text/plain: 




==== Parameters

===== path: thing 

ID of thing to post

- type:  number
- style: simple






==== Response

==== 200
no error



- application/json: array[string]

- text/plain: array[integer]




==== 400
bad parameter




==== 500
server error









#pagebreak()
#show: appendix

= Schemas


// required schema. Don't list them, but do a lookup and mark the field as required
